tests/test_json_s3_to_parquet.py

```diff
@@ -0,0 +1,425 @@
+                     glue_flat_table_name, glue_nested_table, glue_nested_table_name,
+                     glue_crawler_role, namespace):
+        glue_client = boto3.client("glue")
+        crawler_name = "{namespace}-pytest-crawler"
+        time.sleep(10) # give time for the IAM role trust policy to set in
+        glue_crawler = glue_client.create_crawler(
+                Name=crawler_name,
+                Role=glue_crawler_role,
+                DatabaseName=glue_database_name,
+                Description="A crawler for pytest unit test data.",
+                Targets={
+                    "S3Targets": [
+                        {
+                            "Path": os.path.join(
+                                glue_database_path,
+                                glue_flat_table_name.replace("_", "=", 1)
+                            ) + "/"
+                        },
+                        {
+                            "Path": os.path.join(
+                                glue_database_path,
+                                glue_nested_table_name.replace("_", "=", 1)
+                            ) + "/"
+                        }
+                    ]
+                },
+                SchemaChangePolicy={
+                    "DeleteBehavior": "LOG",
+                    "UpdateBehavior": "LOG"
+                },
+                RecrawlPolicy={
+                    "RecrawlBehavior": "CRAWL_NEW_FOLDERS_ONLY"
+                },
+                Configuration=json.dumps({
+                    "Version":1.0,
+                    "CrawlerOutput": {
+                        "Partitions": {
+                            "AddOrUpdateBehavior":"InheritFromTable"
+                        }
+                    },
+                    "Grouping": {
+                        "TableGroupingPolicy":"CombineCompatibleSchemas"
+                    }
+                })
+        )
+        glue_client.start_crawler(Name=crawler_name)
+        response = {"Crawler": {}}
+        for i in range(60): # wait up to 10 minutes for crawler to finish
+            # This should take about 5 minutes
+            response = glue_client.get_crawler(Name=crawler_name)
+            if (
+                    "LastCrawl" in response["Crawler"]
+                    and "Status" in response["Crawler"]["LastCrawl"]
+                    and response["Crawler"]["LastCrawl"]["Status"] == "SUCCEEDED"
+               ):
+                break
+            else:
+                time.sleep(10)
+        yield glue_crawler
+        glue_client.delete_crawler(Name=crawler_name)
+
+    @pytest.fixture()
+    def json_s3_objects(self, datadir, artifact_bucket, namespace):
+        s3_client = boto3.client("s3")
+        dataset_prefix = os.path.join(
+                "BridgeDownstream", namespace, "tests")
+        object_keys = []
+        for dirpath, _, filenames in os.walk(datadir):
+            if len(filenames) > 0:
```

Comment:
Cool trick:  you can just do

```
if filenames:
```
That does the same thing as the `len(filenames) > 0`

Reply:
Nice. I'm going to leave it more explicit.